Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -1,95 +1,171 @@
+            if (IsNullOrEmpty(array) || !CheckRange(array, index))
+            {
+                return array;
+            }
+            else
+            {
+                int[] removeElement = new int[array.Length - 1];
+                for(int i = 0; i < removeElement.Length; i++)
+                {
+                    if (i < index)
+                    {
+                        removeElement[i] = array[i];
+                    }
+                    else
+                    {
+                        removeElement[i] = array[i + 1];
+                    }
+                }
+                return removeElement;
+            }
         }
 
-        /// <summary>
-        /// Inserts a new array element at the start.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <returns>A new array with element added at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertFirst(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            if (IsNullOrEmpty(array))
+            {
+                return new[] { number };
+            }
+            else return InsertAt(array, number, 0);
         }
 
-        /// <summary>
-        /// Inserts a new array element at the end.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <returns>A new array with element added in the end of array. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertLast(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            if (IsNullOrEmpty(array))
+            {
+                return new[] { number };
+            }
+            else return InsertAt(array, number, array.Length);
         }
 
-        /// <summary>
-        /// Inserts a new array element at the specified index.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <param name="index">Index at which array element should be added.</param>
-        /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (CheckForInsert(array, index, false))
+            {
+                return new int[] { number };
+            }
+            else if (!CheckForInsert(array, index, true))
+            {
+                return array;
+            }
+            else
+            {
+                int[] addElement = new int[array.Length + 1];
```

Comment:
``elementsWithInsertion`` or similar should be the name. ``AddElement`` is a verb and a variable name should be a noun.